lib/oban/queue/executor.ex

```diff
@@ -36,9 +36,35 @@ defmodule Oban.Queue.Executor do
   end
 
   @doc false
-  def safe_call(%Job{args: args, worker: worker} = job) do
+  def safe_call(%Job{} = job) do
+    with {:ok, worker} <- worker_module(job) do
+      case worker.timeout(job) do
+        :infinity ->
+          perform_inline(worker, job)
+
+        timeout when is_integer(timeout) ->
+          perform_task(worker, job, timeout)
+      end
+    end
+  end
+
+  defp perform_task(worker, job, timeout) do
+    task = Task.async(fn -> perform_inline(worker, job) end)
+
+    case Task.yield(task, timeout) || Task.shutdown(task) do
+      # reply here could still be a job failure, or even an exception tuple
+      # it just indicates that the job didn't timeout
+      {:ok, reply} ->
+        reply
+
+      nil ->
+        {:current_stacktrace, stacktrace} = Process.info(self(), :current_stacktrace)
```

Comment:
The current stacktrace isn't super helpful here, maybe I should try to take it off the task pid? Not sure about the implications of that.

Reply:
The stacktrace is only important for consistency, I don't think there is any meaningful information that we'd get out of the task either.